### PR TITLE
Added rabbitmq type

### PR DESCRIPTION
**Pull request by evalsocket**
Signed-off-by: Yuvraj <evalsocket@gmail.com>

**What type of PR is this?**

> kind feature

**Any specific area of the project related to this PR?**

> area outputs

**What this PR does / why we need it**: Added new output type RabbitMQ

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -142,6 +142,9 @@ func getConfig() *types.Configuration {
 	v.SetDefault("Kubeless.Kubeconfig", "")
 	v.SetDefault("Kubeless.MinimumPriority", "")
 	v.SetDefault("Webui.URL", "")
+	v.SetDefault("Rabbitmq.MinimumPriority", "")
+	v.SetDefault("Rabbitmq.URL", "")
+	v.SetDefault("Rabbitmq.Queue", "")
```
Comment (Issif) [inline config.go:147]:
```suggestion
	v.SetDefault("Rabbitmq.URL", "")
	v.SetDefault("Rabbitmq.Queue", "")
	v.SetDefault("Rabbitmq.MinimumPriority", "")
```

For keeping same order than others

File: main.go [hunk 2]
```diff
@@ -369,6 +370,16 @@ func init() {
 		}
 	}
 
+	if config.Rabbitmq.URL != "" && config.Rabbitmq.Queue != "" {
+		var err error
+		rabbitmqClient, err = outputs.NewRabbitmqClient( config, stats, promStats, statsdClient, dogstatsdClient)
```
Comment (Issif) [inline main.go:375]:
```suggestion
		rabbitmqClient, err = outputs.NewRabbitmqClient(config, stats, promStats, statsdClient, dogstatsdClient)
```

File: main.go [hunk 3]
```diff
@@ -369,6 +370,16 @@ func init() {
...
+		if err != nil {
+			config.Rabbitmq.URL = ""
+		} else {
+			outputs.EnabledOutputs = append(outputs.EnabledOutputs, "Rabbitmq")
```
Comment (Issif) [inline main.go:379]:
```suggestion
			outputs.EnabledOutputs = append(outputs.EnabledOutputs, "RabbitMQ")
```

File: outputs/rabbitmq.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	"errors"
+	"log"
+	"encoding/json"
+	"github.com/streadway/amqp"
+	"github.com/falcosecurity/falcosidekick/types"
+)
+
+// NewRabbitmqClient returns a new output.Client for accessing the Rabbitmqs API.
```
Comment (Issif) [inline outputs/rabbitmq.go:11]:
```suggestion
// NewRabbitmqClient returns a new output.Client for accessing the RabbitmMQ API.
```

File: outputs/rabbitmq.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+func NewRabbitmqClient(config *types.Configuration, stats *types.Statistics, promStats *types.PromStatistics, statsdClient, dogstatsdClient *statsd.Client) (*Client, error) {
+
+	var channel *amqp.Channel
+	if config.Rabbitmq.URL != "" && config.Rabbitmq.Queue != "" {
+		conn, err := amqp.Dial(config.Rabbitmq.URL)
+		if err != nil {
+			log.Printf("[ERROR] : Rabbitmq - %v\n", "Error while connecting rabbitmq")
```
Comment (Issif) [inline outputs/rabbitmq.go:18]:
```suggestion
			log.Printf("[ERROR] : RabbitMQ - %v\n", "Error while connecting rabbitmq")
```

File: outputs/rabbitmq.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		ch, err := conn.Channel()
+		if err != nil {
+			log.Printf("[ERROR] : Rabbitmq Channel - %v\n", "Error while creating rabbitmq channel")
+			return nil, errors.New("Error while creating rabbitmq channel")
+		}
+		channel = ch
+	}
+
+	return &Client{
+		OutputType:      "GCP",
```
Comment (Issif) [inline outputs/rabbitmq.go:30]:
```suggestion
		OutputType:      "RabbitMQ",
```

File: outputs/rabbitmq.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	}, nil
+}
+
+// Publish sends a message to a Rabbitmq
+func (c *Client) Publish(falcopayload types.FalcoPayload) {
+	c.Stats.Rabbitmq.Add(Total, 1)
+
+	payload, _ := json.Marshal(falcopayload)
+
+	err := c.RabbitmqClient.Publish("", c.Config.Rabbitmq.Queue, false,false, amqp.Publishing {
```
Comment (Issif) [inline outputs/rabbitmq.go:46]:
```suggestion
	err := c.RabbitmqClient.Publish("", c.Config.Rabbitmq.Queue, false, false, amqp.Publishing {
```

File: outputs/rabbitmq.go [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+	if err != nil {
+		log.Printf("[ERROR] : RabbitMQ - %v - %v\n", "Error while publishing message", err.Error())
+		c.Stats.Rabbitmq.Add(Error, 1)
+		go c.CountMetric("outputs", 1, []string{"output:rabbitmq", "status:error"})
+		c.PromStats.Outputs.With(map[string]string{"destination": "rabbitmq", "status": Error}).Inc()
+
+		return
+	}
+
+	log.Printf("[INFO]  : rabbitmq - Send to message OK \n")
```
Comment (Issif) [inline outputs/rabbitmq.go:60]:
```suggestion
	log.Printf("[INFO]  : RabbitMQ - Send to message OK\n")
```

**Review by Issif — CHANGES_REQUESTED** [review]
you have some linting issue to fix 😉

**Review by Issif — APPROVED** [review]

**Review by leogr — COMMENTED** [review]
Hey,

this PR LGTM, thank you! :+1: 

However, there're merge commits which are not allowed in this repo, and all commits need to be [signed-off](https://github.com/falcosecurity/.github/blob/master/CONTRIBUTING.md#developer-certificate-of-origin).

You can fix both with the following one-liner:
```
git rebase -i --exec 'git commit --amend --no-edit -n -s -S' master
```
(and resolve conflicts, if any)
Finally, you can `git push --force-with-lease` your branch to update this PR.


Thanks in advance, and let me know if you need any support with that.

**Review by Issif — APPROVED** [review]

**Pull request merged**